### DISCUSSION TITLE
[Suggestion] MoreCreativeTabs Compat

### DISCUSSION THREAD

**Discussion by CornerBasket** [melontini/recipe_book_is_pain#20] [Ideas]
**Is your feature request related to a problem? Please describe.**
I was hoping to use MoreCreativeTabs in conjunction with this mod to design custom recipe book tabs, but unfortunately it doesn't work right out the gate.

**Describe the solution you'd like**
Add compatibility with MCT's resourcepack-driven customized tabs so that they're "copied" correctly after reloading.

**Describe alternatives you've considered**
If compatibility isn't an option, would you consider adding a similar system for modifying creative/recipe tabs? Perhaps via datapack?

**Comment by zenfyrdev** [comment]
Hi, thanks for the suggestion

Despite what it may seem like, recipe book groups are nothing like item groups codewise. So, I can't think of a good way to support MoreCreativeTabs.

As per the data-driven system you suggest. This sounds like an interesting idea, but the last time I tried it, I ran into the problem of the server refusing to send tags to the client, which broke the whole thing when playing on dedicated servers. This will also have to be a separate mod as this is too out-of-scope for RBIP.

There's also the implementation issue. If with item groups, you get a list of item stacks where you can add/remove anything. With recipe book groups, you get an `Iterable` of recipes and from that `Iterable` you have to make a map of lists, of lists. (This is not a joke, btw: `Map<RecipeBookGroup, List<List<Recipe<?>>>>`).

And there's also another thing: recipe book groups are java enums and don't have proper identifiers.

And the best part is: I could theoretically make the crafting table show furnace recipes, since categories and groups are not linked to recipe types in any way.

So this could probably look something like this:

In a config file:
- You create new groups.

```json
    "groups": {
        "MYID_MYGROUP": {
            "category": "CRAFTING",
            "index": 3,
            "icons": [
                {
                    "item": "minecraft:coal"
                },
                {
                    "item": "modid:whatever"
                }
            ]
        }
    },
```

- You create recipe lookups based on recipe/type/output predicates.

```json
    "lookups": [
        {
            "type": "crafting_shaped",
            "recipe": {
                "regex": "insert id regex here"
            },
            "output": {
                "regex": "insert id regex here",
                "item":"minecraft:coal",
                "item_group": "minecraft:building_blocks"
            },
            "group": "MYID_MYGROUP"
        }
    ]
```
